### PR TITLE
Add doc for counting categorical dtype

### LINKED ISSUE #40735: BUG (?): dtype.value_counts() shows categorical multiple times
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
>>> df = pd.DataFrame({'a': [1], 'b': ['2'], 'c': [3], 'd': [3]}).astype({'a': 'category', 'c': 'category', 'd': 'category'})
>>> df
   a  b  c  d
0  1  2  3  3
>>> df.dtypes.value_counts()
category    2
category    1
object      1
dtype: int64

```

#### Problem description

`category` appears twice with different counts

#### Expected Output

Either
```python
category    3
object      1
dtype: int64
```
or
```python
CategoricalDtype(categories=[3], ordered=False)    2
CategoricalDtype(categories=[1], ordered=False)    1
object      1
dtype: int64


```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 8064973159c46d3c4608e3aaf637196e224eba91
python           : 3.8.6.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-70-generic
Version          : #78-Ubuntu SMP Fri Mar 19 13:29:52 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 1.3.0.dev0+1211.g8064973159
numpy            : 1.19.5
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 20.3.3
setuptools       : 49.6.0.post20201009
Cython           : 0.29.22
pytest           : 6.2.2
hypothesis       : 6.8.1
sphinx           : 3.5.2
blosc            : None
feather          : None
xlsxwriter       : 1.3.7
lxml.etree       : 4.6.2
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.19.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 0.8.7
fastparquet      : 0.5.0
gcsfs            : 0.7.2
matplotlib       : 3.3.3
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : 2.0.0
pyxlsb           : None
s3fs             : 0.5.2
scipy            : 1.6.1
sqlalchemy       : 1.4.2
tables           : 3.6.1
tabulate         : 0.8.9
xarray           : 0.17.0
xlrd             : 2.0.1
xlwt             : 1.3.0
numba            : 0.52.0

</details>

### PR DESCRIPTION
**Pull request by jahn96**
- [x] closes #40735
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: doc/source/user_guide/categorical.rst [hunk 1]
```diff
@@ -1178,3 +1180,17 @@ Use ``copy=True`` to prevent such a behaviour or simply don't reuse ``Categorica
...
+Counting CategoricalDtype
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As mentioned in :ref:`Equality Semantics <categorical.equalitysemantics>`, two instances of :class:`~pandas.api.types.CategoricalDtype` compare equal
+whenever they have the same categories and order. Therefore, when counting data types, the multiple instances of :class:`~pandas.api.types.CategoricalDtype` will be counted as one group if they have the same categories and order.
+In the example below, even though ``a``, ``c``, and ``d`` all have data types of ``category``, they will not be counted as one group since they don't have the same categories.
+
+.. ipython:: python
+
+    df = pd.DataFrame({'a': [1], 'b': ['2'], 'c': [3], 'd': [3]}).astype({'a': 'category', 'c': 'category', 'd': 'category'})
```
Comment (mroeschke) [inline doc/source/user_guide/categorical.rst:1193]: Can you put this in the docstring instead?

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/categorical.rst (hunk 1 above)
Comment (jahn96) [reply doc/source/user_guide/categorical.rst]: addressed [here](https://github.com/pandas-dev/pandas/pull/59327/commits/1348bdc09df6480a0a6108a2bcc21848e04da6d0)

File: doc/source/user_guide/categorical.rst [hunk 2]
```diff
@@ -1178,3 +1180,9 @@ Use ``copy=True`` to prevent such a behaviour or simply don't reuse ``Categorica
     This also happens in some cases when you supply a NumPy array instead of a ``Categorical``:
     using an int array (e.g. ``np.array([1,2,3,4])``) will exhibit the same behavior, while using
     a string array (e.g. ``np.array(["a","b","c","a"])``) will not.
+
+Counting CategoricalDtype
```
Comment (mroeschke) [inline doc/source/user_guide/categorical.rst:1184]: Could you undo the changes in this file?

File: pandas/core/base.py [hunk 3]
```diff
@@ -1049,6 +1049,34 @@ def value_counts(
         4.0    1
         NaN    1
         Name: count, dtype: int64
+
+        **categorial_dtypes**
```
Comment (mroeschke) [inline pandas/core/base.py:1053]:
```suggestion
        **Categorial Dtypes**
```

File: doc/source/user_guide/categorical.rst (hunk 2 above)
Comment (jahn96) [reply doc/source/user_guide/categorical.rst:1184]: Addressed [here](https://github.com/pandas-dev/pandas/pull/59327/commits/d8381812e305c6c60f5fde26ded4b907fe96db04)

File: pandas/core/base.py (hunk 3 above)
Comment (jahn96) [reply pandas/core/base.py]: Addressed [here](https://github.com/pandas-dev/pandas/pull/59327/commits/86facac3c582c399e600a7a0f0f6cb7d26495760)

File: doc/source/user_guide/categorical.rst (hunk 2 above)
Comment (mroeschke) [reply doc/source/user_guide/categorical.rst:1184]: Looks like there's still changes in this file

Comment (jahn96) [reply doc/source/user_guide/categorical.rst]: Ah sorry! Addressed [here](https://github.com/pandas-dev/pandas/pull/59327/commits/a847c134641a6e5dfa1c95e2e30bd6e67885b8cf)

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jahn96
